File: app/src/main/java/com/tammamkhalaf/myuaeguide/categories/hereDeveloper/discoverHere/LockupEntrypoint/PlaceResponse.java

package com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

@Entity(tableName = "PlacesTable")
public class PlaceResponse{

    @Expose(deserialize = false, serialize = false)
    @PrimaryKey(autoGenerate = true)
    private int id;

    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("placeId")
    @Expose
    private String placeId;
    @SerializedName("view")
    @Expose
    private String view;

    @SerializedName("icon")
    @Expose
    private String icon;

    @Ignore
    @SerializedName("location")
    @Expose
    private Location location;

    @Ignore
    @SerializedName("contacts")
    @Expose
    private Contacts contacts;

    @Ignore
    @SerializedName("categories")
    @Expose
    private List<Category> categories;

    @Ignore
    @SerializedName("media")
    @Expose
    private Media media;


    @SerializedName("extended")
    @Expose
    @Ignore
    private Extended extended;

    @Ignore
    @SerializedName("related")
    @Expose
    private Related related;

    public PlaceResponse(String name, String placeId, String view, Location location, Contacts contacts,
                         List<Category> categories, String icon, Media media, Extended extended, Related related) {
        this.name = name;
        this.placeId = placeId;
        this.view = view;
        this.location = location;
        this.contacts = contacts;
        this.categories = categories;
        this.icon = icon;
        this.media = media;
        this.extended = extended;
        this.related = related;
    }

    public PlaceResponse() {
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPlaceId() {
        return placeId;
    }

    public void setPlaceId(String placeId) {
        this.placeId = placeId;
    }

    public String getView() {
        return view;
    }

    public void setView(String view) {
        this.view = view;
    }

    public Location getLocation() {
        return location;
    }

    public void setLocation(Location location) {
        this.location = location;
    }

    public Contacts getContacts() {
        return contacts;
    }

    public void setContacts(Contacts contacts) {
        this.contacts = contacts;
    }

    public List<Category> getCategories() {
        return categories;
    }

    public void setCategories(List<Category> categories) {
        this.categories = categories;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public Media getMedia() {
        return media;
    }

    public void setMedia(Media media) {
        this.media = media;
    }

    public Extended getExtended() {
        return extended;
    }

    public void setExtended(Extended extended) {
        this.extended = extended;
    }

    public Related getRelated() {
        return related;
    }

    public void setRelated(Related related) {
        this.related = related;
    }

    public int getId() {
        return id;
    }
}

File: kmmsharedmodule/src/commonMain/kotlin/com/tammamkhalaf/kmmsharedmodule/categories/hereDeveloper/discoverHere/LockupEntrypoint/PlaceResponse.kt
package com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint

import androidx.room.Entity

@Entity(tableName = "PlacesTable")
class PlaceResponse {
    @Expose(deserialize = false, serialize = false)
    @PrimaryKey(autoGenerate = true)
    private var id = 0

    @SerializedName("name")
    @Expose
    private var name: String? = null

    @SerializedName("placeId")
    @Expose
    private var placeId: String? = null

    @SerializedName("view")
    @Expose
    private var view: String? = null

    @SerializedName("icon")
    @Expose
    private var icon: String? = null

    @Ignore
    @SerializedName("location")
    @Expose
    private var location: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Location? = null

    @Ignore
    @SerializedName("contacts")
    @Expose
    private var contacts: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Contacts? = null

    @Ignore
    @SerializedName("categories")
    @Expose
    private var categories: List<com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Category>? = null

    @Ignore
    @SerializedName("media")
    @Expose
    private var media: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Media? = null

    @SerializedName("extended")
    @Expose
    @Ignore
    private var extended: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Extended? = null

    @Ignore
    @SerializedName("related")
    @Expose
    private var related: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Related? = null

    constructor(name: String?, placeId: String?, view: String?, location: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Location?, contacts: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Contacts?,
                categories: List<com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Category>?, icon: String?, media: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Media?, extended: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Extended?, related: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Related?) {
        this.name = name
        this.placeId = placeId
        this.view = view
        this.location = location
        this.contacts = contacts
        this.categories = categories
        this.icon = icon
        this.media = media
        this.extended = extended
        this.related = related
    }

    constructor() {}

    fun setId(id: Int) {
        this.id = id
    }

    fun getName(): String? {
        return name
    }

    fun setName(name: String?) {
        this.name = name
    }

    fun getPlaceId(): String? {
        return placeId
    }

    fun setPlaceId(placeId: String?) {
        this.placeId = placeId
    }

    fun getView(): String? {
        return view
    }

    fun setView(view: String?) {
        this.view = view
    }

    fun getLocation(): com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Location? {
        return location
    }

    fun setLocation(location: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Location?) {
        this.location = location
    }

    fun getContacts(): com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Contacts? {
        return contacts
    }

    fun setContacts(contacts: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Contacts?) {
        this.contacts = contacts
    }

    fun getCategories(): List<com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Category>? {
        return categories
    }

    fun setCategories(categories: List<com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Category>?) {
        this.categories = categories
    }

    fun getIcon(): String? {
        return icon
    }

    fun setIcon(icon: String?) {
        this.icon = icon
    }

    fun getMedia(): com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Media? {
        return media
    }

    fun setMedia(media: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Media?) {
        this.media = media
    }

    fun getExtended(): com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Extended? {
        return extended
    }

    fun setExtended(extended: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Extended?) {
        this.extended = extended
    }

    fun getRelated(): com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Related? {
        return related
    }

    fun setRelated(related: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Related?) {
        this.related = related
    }

    fun getId(): Int {
        return id
    }
}